src/decoders/ascii85_decoder.rs

```diff
@@ -0,0 +1,176 @@
+/// let athena_checker = Checker::<Athena>::new();
+/// let checker = CheckerTypes::CheckAthena(athena_checker);
+///
+/// let result = decode_ascii85.crack("BOu!rD]j7BEbo7", &checker).unencrypted_text;
+/// assert!(result.is_some());
+/// assert_eq!(result.unwrap()[0], "hello world");
+/// ```
+pub struct ASCII85Decoder;
+
+impl Crack for Decoder<ASCII85Decoder> {
+    fn new() -> Decoder<ASCII85Decoder> {
+        Decoder {
+            name: "ASCII85",
+            description: "Ascii85, also called Base85, is a form of binary-to-text encoding that uses five ASCII characters to represent four bytes of binary data.",
+            link: "https://en.wikipedia.org/wiki/Ascii85",
+            tags: vec!["decoder", "ascii85"],
+            popularity: 1.0,
+            phantom: std::marker::PhantomData,
+        }
+    }
+
+    /// This function does the actual decoding
+    /// It returns an Option<string> if it was successful
+    /// Else the Option returns nothing and the error is logged in Trace
+    fn crack(&self, text: &str, checker: &CheckerTypes) -> CrackResult {
+        trace!("Trying ASCII85 with text {:?}", text);
+        let decoded_text = decode_ascii85_no_error_handling(text);
+        let mut results = CrackResult::new(self, text.to_string());
+
+        if decoded_text.is_none() {
+            debug!("Failed to decode ascii85 because ASCII85Decoder::decode_ascii85_no_error_handling returned None");
+            return results;
+        }
+
+        let decoded_text = decoded_text.unwrap();
+        if !check_string_success(&decoded_text, text) {
+            info!(
+                "Failed to decode ascii85 because check_string_success returned false on string {}",
+                decoded_text
+            );
+            return results;
+        }
+
+        let checker_result = checker.check(&decoded_text);
+        results.unencrypted_text = Some(vec![decoded_text]);
+
+        results.update_checker(&checker_result);
+
+        results
+    }
+    /// Gets all tags for this decoder
+    fn get_tags(&self) -> &Vec<&str> {
+        &self.tags
+    }
+    /// Gets the name for the current decoder
+    fn get_name(&self) -> &str {
+        self.name
+    }
+}
+
+/// helper function
+fn decode_ascii85_no_error_handling(text: &str) -> Option<String> {
+    // Runs the code to decode ascii85
+    // Doesn't perform error handling, call from_ascii85
+    ascii85::decode(text)
+        .ok()
+        .map(|inner| String::from_utf8(inner).ok())?
+}
+
+#[cfg(test)]
+mod tests {
+    use super::ASCII85Decoder;
+    use crate::{
+        checkers::{
+            athena::Athena,
+            checker_type::{Check, Checker},
+            CheckerTypes,
+        },
+        decoders::interface::{Crack, Decoder},
+    };
+
+    // helper for tests
+    fn get_athena_checker() -> CheckerTypes {
+        let athena_checker = Checker::<Athena>::new();
+        CheckerTypes::CheckAthena(athena_checker)
+    }
+
+    #[test]
```

Comment:
Add more tests for base85 vs ascii85. There should be tests to make sure this only decodes ascii85 and not base85. Also add tests with longer strings and special characters in the plaintext. See the base64 URL tests for an example.